File: chapter02/src/test/java/main/MainTest.java
package main;

import static org.junit.Assert.*;
import static org.junit.Assume.*;

import static org.hamcrest.CoreMatchers.*;

import java.util.Arrays;
import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;

@FixMethodOrder(MethodSorters.DEFAULT)
public class MainTest {

  private Main main = new Main();

  @Test
  public void usualLogic() {
    assertThat(main.usualLogic(true, true, false), equalTo(true));
  }

  @Test
  public void unUsualLogic() {
    assertThat(Arrays.asList("1", "3"), hasItem("1"));
    assertThat("xyy", startsWith("x"));
  }

  @Test
  public void nestedLogic() {
  }
}

File: chapter02/src/test/kotlin/main/MainTest.kt
package main

import org.hamcrest.CoreMatchers.*
import org.hamcrest.MatcherAssert.assertThat
import org.junit.Test

class MainTestKt {

    private val main = Main()

    @Test
    fun usualLogic() {
        assertThat(main.usualLogic(true, true, false), equalTo(true))
    }

    @Test
    fun unUsualLogic() {
        assertThat(listOf("1", 2), hasItem(2))
    }

    @Test
    fun nestedLogic() {
    }
}